src/main/java/duke/MrMoon.java

```diff
@@ -0,0 +1,137 @@
+package duke;
+
+import duke.command.ClearCommand;
+import duke.command.Command;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.ui.Ui;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Main Duke application class that coordinates all components.
+ * Handles the main program loop, user input processing, and command execution.
+ * Manages the interaction between UI, Parser, TaskList, and Storage components.
+ */
+public class MrMoon {
+
+    /** The task list containing all user tasks */
+    private final TaskList tasks;
+
+    /** The user interface component for input/output */
+    private final Ui ui;
+
+    /** The command parser for interpreting user input */
+    private final Parser parser;
+
+    /** The scanner for reading user input */
+    private final Scanner scanner;
+
+    /**
+     * Constructs the main Duke application with the specified storage file path.
+     * Initializes all components and loads existing tasks from storage.
+     *
+     * @param filePath The file path for task data storage
+     */
+    public MrMoon(String filePath) {
+        this.ui = new Ui(System.out);
+        this.parser = new Parser();
+        Storage storage = new Storage(filePath);
+        this.scanner = new Scanner(System.in);
+
+        List<Task> loaded;
```

Comment:
Is there a better name to describe this variable?
```suggestion
        List<Task> loadedTasks;
```

Reply:
Thanks man!